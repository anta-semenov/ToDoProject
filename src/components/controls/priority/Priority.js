import React from 'react';

export default class Priority extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (<div>MyComponent</div>);
  }
}

Priority.propTypes = {
};
