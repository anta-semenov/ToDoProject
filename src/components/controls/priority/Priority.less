@import '../../../styles/variables.less';

.priority {
  position: relative;
  z-index: 2;
  box-sizing: border-box;

  cursor: pointer;
  transform-origin: center left;
  transition: 0.2s ease-out;
  flex-shrink: 0;

  &:hover {
    z-index: 10;
    transform: scale(4.5);// translateX(-2px);
    background-color: #ffffff;
    box-shadow: 0 2px 8px 0 fade(#000000, 20%);
    border-radius: 1px;
    transition: 0.3s ease-out 0.15s;
  }

  &.is-disabled {
    pointer-events: none;
    opacity: 0.5;
  }
  &.is-dimmed {
    opacity: 0.5;
  }
  &.priority--default {
    margin: 0;
    padding: 4px;
    width: 22px;
    height: 22px;
  }
  &.priority--tasks-list {
    margin: 0 10px 0 0;
    padding: 4px;
    width: 22px;
    height: 22px;
  }
  &.priority--task-info {
    margin: 0;
    padding: 4px;
    width: 22px;
    height: 22px;
    transform-origin: left top;
  }
}
.priority-level {
  position: relative;
  z-index: 1;
  height: 2px;
  margin: 0 0 2px 0;
  background-color: @color-border-gray;
  transition: @transition-basic;

  &:hover {
    background-color: @color-border-dark-gray;
  }

  &.priority-level--none {
    position: absolute;
    z-index: 0;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: transparent;
  }

  &.priority-level--low {
    width: 8px;
    .priority.priority--low & {
      background-color: @color-accent;
    }
  }

  &.priority-level--medium {
    width: 10px;
    .priority.priority--medium & {
      background-color: @color-accent;
    }
  }

  &.priority-level--high {
    width: 12px;
    .priority.priority--high & {
      background-color: @color-accent;
    }
  }

  &.priority-level--max {
    width: 14px;
    .priority.priority--max & {
      background-color: @color-accent;
    }
  }
}
