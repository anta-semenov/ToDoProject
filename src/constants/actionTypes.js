/*
 * Project
 */
export const ADD_PROJECT = 'ADD_PROJECT'
export const REMOVE_PROJECT = 'REMOVE_PROJECT'
export const COMPLETE_PROJECT = 'COMPLETE_PROJECT'
export const EDIT_PROJECT = 'EDIT_PROJECT'
export const REPLACE_PROJECT = 'REPLACE_PROJECT'

/*
 * Task
 */
export const ADD_TASK = 'ADD_TASK'
export const REMOVE_TASK = 'REMOVE_TASK'
export const COMPLETE_TASK = 'COMPLETE_TASK'
export const EDIT_TASK = 'EDIT_TASK'
export const REPLACE_TASK = 'REPLACE_TASK'
export const ADD_TASK_TO_PROJECT = 'ADD_TASK_TO_PROJECT'
export const ADD_TASK_CONTEXT = 'ADD_TASK_CONTEXT'
export const REMOVE_TASK_CONTEXT = 'REMOVE_TASK_CONTEXT'
export const SET_TASK_TODAY = 'SET_TASK_TODAY'
export const START_TASK_TRACKING = 'START_TASK_TRACKING'
export const STOP_TASK_TRACKING = 'STOP_TASK_TRACKING'

/*
 * Context
 */
export const ADD_CONTEXT = 'ADD_CONTEXT'
export const REMOVE_CONTEXT = 'REMOVE_CONTEXT'
export const EDIT_CONTEXT = 'EDIT_CONTEXT'
export const SWITCH_TASK_CONTEXT = 'SWITCH_TASK_CONTEXT'
export const REPLACE_CONTEXT = 'REPLACE_CONTEXT'

/*
* UI State
*/
export const SET_SELECTED_SECTION = 'SET_SELECTED_SECTION'
export const SET_SIDEBAR_SIZE = 'SET_SIDEBAR_SIZE'
export const SET_ACTIVE_ITEM = 'SET_ACTIVE_ITEM'
export const SET_EDITING_SECTION = 'SET_EDITING_SECTION'
export const TOGGLE_TASK_COMPLETED_LATENCY = 'TOGGLE_TASK_COMPLETED_LATENCY'
export const CLEAR_COMPLETED_LATENT_TASKS = 'CLEAR_COMPLETED_LATENT_TASKS'
export const TOGGLE_TASK_LATENCY = 'TOGGLE_TASK_LATENCY'
export const CLEAR_LATENT_TASKS = 'CLEAR_LATENT_TASKS'
export const SET_SYNCING = 'SET_SYNCING'
export const SET_OFFLINE = 'SET_OFFLINE'
export const SET_AUTH_STATUS = 'SET_AUTH_STATUS'
export const SET_AUTH_ERROR_MESSAGE = 'SET_AUTH_ERROR_MESSAGE'
export const SET_PROPERTY = 'SET_PROPERTY'

/*
* Common
*/
export const SET_STATE = 'SET_STATE'
export const LOG_OUT = 'LOG_OUT'
export const REQUEST_AUTH = 'REQUEST_AUTH'
export const RECIEVE_AUTH = 'RECIEVE_AUTH'
export const ERROR_AUTH = 'ERROR_AUTH'

export const REQUEST_DATA = 'REQUEST_DATA'
export const RECIEVE_DATA = 'RECIEVE_DATA'
export const ERROR_DATA = 'ERROR_DATA'

export const UNDO = 'UNDO'
export const REDO = 'REDO'
export const CLEAR_UNDO_REDO = 'CLEAR_UNDO_REDO'
