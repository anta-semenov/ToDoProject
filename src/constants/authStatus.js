export const AUTH_IN_PROGRESS = 'AUTH_IN_PROGRESS'
export const AUTH_SUCESS = 'AUTH_SUCESS'
export const AUTH_ERROR = 'AUTH_ERROR'
export const AUTH_NONE = 'AUTH_NONE'
