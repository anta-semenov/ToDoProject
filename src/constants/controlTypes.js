export const TASK_TITLE_FIELD = 'TASK_TITLE_FIELD'
export const TASK_DESCRIPTION_FIELD = 'TASK_DESCRIPTION_FIELD'
