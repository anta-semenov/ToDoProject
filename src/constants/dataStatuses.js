export const DATA_NONE = 'DATA_NONE'
export const DATA_REQUESTED = 'DATA_REQUESTED'
export const DATA_ERROR = 'DATA_ERROR'
export const DATA_RECIEVED = 'DATA_RECIEVED'
