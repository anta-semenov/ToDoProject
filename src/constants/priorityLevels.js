export const PRIORITY_NONE = 'none'
export const PRIORITY_LOW = 'low'
export const PRIORITY_MEDIUM = 'medium'
export const PRIORITY_HIGH = 'high'
export const PRIORITY_MAX = 'max'
export const PRIORITY = [PRIORITY_NONE, PRIORITY_LOW, PRIORITY_MEDIUM, PRIORITY_HIGH, PRIORITY_MAX]
