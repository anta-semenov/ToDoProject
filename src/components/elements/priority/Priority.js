import PropTypes from 'prop-types';
import React from 'react'
import classNames from 'classnames'
import { PRIORITY_NONE, PRIORITY_MAX, PRIORITY_HIGH, PRIORITY_MEDIUM, PRIORITY_LOW } from '../../../constants/priorityLevels'
import './Priority.less'

const Priority = ({ priority = PRIORITY_NONE, appearance = 'default', disabled = false, dimmed = false, onClick }) => {
  const priorityClasses = classNames({
    'priority': true,
    [`priority--${priority}`]: true,
    [`priority--${appearance}`]: true,
    'is-dimmed': dimmed,
    'is-disabled': disabled
  })
  return (
    <div className={priorityClasses} >
      <div className='priority-level priority-level--none' onClick={() => onClick(PRIORITY_NONE)} />
      <div className='priority-level priority-level--max' onClick={() => onClick(PRIORITY_MAX)} />
      <div className='priority-level priority-level--high' onClick={() => onClick(PRIORITY_HIGH)} />
      <div className='priority-level priority-level--medium' onClick={() => onClick(PRIORITY_MEDIUM)} />
      <div className='priority-level priority-level--low' onClick={() => onClick(PRIORITY_LOW)} />
    </div>
  )
}

Priority.propTypes = {
  priority: PropTypes.oneOf([PRIORITY_NONE, PRIORITY_LOW, PRIORITY_MEDIUM, PRIORITY_HIGH, PRIORITY_MAX]),
  appearance: PropTypes.oneOf(['task-info', 'tasks-list', 'default']),
  disabled: PropTypes.bool,
  dimmed: PropTypes.bool,
  onClick: PropTypes.func.isRequired
}

export default Priority
