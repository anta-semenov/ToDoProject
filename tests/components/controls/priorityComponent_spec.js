import React from 'react'
import { shallow } from 'enzyme'

import * as priorityLevels from '../../../src/constants/priorityLevels'
import Priority from '../../../src/components/elements/priority/Priority'

describe('Priority', () => {
  test('Should render priority component with right classes for default props', () => {
    const closeBtn = shallow(<Priority onClick={jest.fn()} />)
    expect(closeBtn).toMatchSnapshot()
  })
  test('Should render priority with right classes for custom props', () => {
    const closeBtn = shallow(
      <Priority
        appearance="task-info"
        priority={priorityLevels.PRIORITY_MEDIUM}
        disabled={true}
        onClick={jest.fn()}
      />
    )
    expect(closeBtn).toMatchSnapshot()
  })
})
