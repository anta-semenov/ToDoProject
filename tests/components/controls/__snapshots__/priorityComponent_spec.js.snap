// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Priority Should render priority component with right classes for default props 1`] = `
<div
  className="priority priority--none priority--default"
>
  <div
    className="priority-level priority-level--none"
    onClick={[Function]}
  />
  <div
    className="priority-level priority-level--max"
    onClick={[Function]}
  />
  <div
    className="priority-level priority-level--high"
    onClick={[Function]}
  />
  <div
    className="priority-level priority-level--medium"
    onClick={[Function]}
  />
  <div
    className="priority-level priority-level--low"
    onClick={[Function]}
  />
</div>
`;

exports[`Priority Should render priority with right classes for custom props 1`] = `
<div
  className="priority priority--medium priority--task-info is-disabled"
>
  <div
    className="priority-level priority-level--none"
    onClick={[Function]}
  />
  <div
    className="priority-level priority-level--max"
    onClick={[Function]}
  />
  <div
    className="priority-level priority-level--high"
    onClick={[Function]}
  />
  <div
    className="priority-level priority-level--medium"
    onClick={[Function]}
  />
  <div
    className="priority-level priority-level--low"
    onClick={[Function]}
  />
</div>
`;
